module/purpose/ScriptTuner/src/ScriptTuner.cpp

```diff
@@ -141,25 +158,114 @@ namespace module::purpose {
             emit(std::move(waypoint));
         });
 
+        // Debugging to make sure logs are working
+        /// TODO: Remove before merging with main
+        on<Every<5, std::chrono::seconds>>().then("Heartbeat", [this] {
+            static size_t counter = 0;
+            log<NUClear::DEBUG>("Beep Boop ", counter++);
+        });
+
+        on<Every<AUTOSAVE_INTERVAL, std::chrono::seconds>, Single>().then("Autosave", [this] {
+            log<NUClear::DEBUG>("Autosaving script if required...");
+
+            // Only autosave if autosaving is enabled and we have unsaved changes
+            if (!autosave_enabled || !unsaved_changes)
+                return;
+
+            // Before we start, save the temp save the old path so we can delete it
+            // once we have a new autosave file.
+            auto last_autosave_path = this->autosave_path;
+
+            // Split the script name into the path and filename
+            auto [autosave_dir, filename] = utility::file::pathSplit(script_path);
```

Comment:
use the standard library filesystem::path functions, we have a PR up somewhere to remove `fileutil.cpp`.